lib/ansible/modules/system/zfs_permissions.py

```diff
@@ -0,0 +1,271 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# (c) 2015, Nate Coraor <nate@coraor.org>
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+DOCUMENTATION = '''
+---
+module: zfs_permissions
+short_description: Manage zfs administrative permissions
+description:
+  - Manages ZFS file system administrative permissions on Solaris and FreeBSD. See zfs(1M) for more information about the properties.
+version_added: "2.2"
+options:
+  name:
+    description:
+      - File system or volume name e.g. C(rpool/myfs)
+    required: true
+  state:
+    description:
+      - Whether to allow (C(present)), or unallow (C(absent)) a permission.
+    required: true
+    choices: [present, absent]
+  users:
+    description:
+      - Users to whom permission(s) should be granted, separated by commas.
+    required: false
```

Comment:
True, older modules do no comply to the new module contribution documentation because it did not exist. However it is now part of this: http://docs.ansible.com/ansible/dev_guide/developing_modules_checklist.html

BTW It is easier to find which parameters are required, when only the ones that are required have `required: true`. If they all have a `required:` option, you have to find which ones have true on the right line. Not very convenient, hence we have this best practice.